static/main.ts

```diff
@@ -12,30 +13,33 @@ async function handleAuth() {
 
 export async function mainModule() {
   const auth = await handleAuth();
-  const decoder = new ManifestDecoder();
   const renderer = new ManifestRenderer(auth);
-  const search = window.location.search.substring(1);
-
-  if (search) {
-    const decodedManifest = await decoder.decodeManifestFromSearch(search);
-    return renderer.renderManifest(decodedManifest);
-  }
+  renderer.manifestGuiBody.dataset.loading = "false";
 
   try {
     const ubiquityOrgsToFetchOfficialConfigFrom = ["ubiquity-os"];
-    const fetcher = new ManifestFetcher(ubiquityOrgsToFetchOfficialConfigFrom, auth.octokit, decoder);
+    const fetcher = new ManifestFetcher(ubiquityOrgsToFetchOfficialConfigFrom, auth.octokit);
     const cache = fetcher.checkManifestCache();
+
     if (auth.isActiveSession()) {
       const userOrgs = await auth.getGitHubUserOrgs();
-      renderer.renderOrgPicker(userOrgs);
+
       if (Object.keys(cache).length === 0) {
+        renderer.manifestGuiBody.dataset.loading = "true";
+        const killNotification = toastNotification("Fetching manifest data...", { type: "info", shouldAutoDismiss: true });
+        renderOrgPicker(renderer, []);
+
         const manifestCache = await fetcher.fetchMarketplaceManifests();
         localStorage.setItem("manifestCache", JSON.stringify(manifestCache));
```

Comment:
This is a duplicate cache set, you already cache manifests inside `fetchMarketplaceManifests()` on L41 of `fetch-manifest.ts`:

`localStorage.setItem("manifestCache", JSON.stringify(manifestCache));`